comms/discovery/db/queries/get_new_blasts.go

```diff
@@ -28,7 +28,17 @@ func GetNewBlasts(q db.Queryable, ctx context.Context, arg ChatMembershipParams)
 	// see also: subtly different inverse query exists in chat_blast.go
 	// to fan out messages to existing chat
 	var findNewBlasts = `
-	with all_new as (
+	with
+	last_permission_change as (
+		select max(t) as t from (
+			select updated_at as t from chat_permissions where user_id = $1
+			union
+			select created_at as t from chat_blocked_users where blocker_user_id = $1
```

Comment:
this one feels potentially weird but i think it works. ideally we'd select specifically the row where the blocker and blockee match the to/from user ids, but that's hard.
i think this is fine bc by the time the user updates their permissions they would have already upgrade any existing blasts.